src/app/util/ContentApp.cpp

```diff
@@ -0,0 +1,164 @@
+                                              uint16_t maxReadLength)
+{
+    ChipLogProgress(DeviceLayer, "Read Attribute for device %s cluster %d attribute=%d)",
+                    GetApplicationBasic()->GetApplicationName(), static_cast<uint16_t>(clusterId),
+                    static_cast<uint16_t>(attributeId));
+
+    EmberAfStatus ret = EMBER_ZCL_STATUS_FAILURE;
+    if (clusterId == ZCL_APPLICATION_BASIC_CLUSTER_ID)
+    {
+        ret = GetApplicationBasic()->HandleReadAttribute(attributeId, buffer, maxReadLength);
+    }
+    if (clusterId == ZCL_ACCOUNT_LOGIN_CLUSTER_ID)
+    {
+        ret = GetAccountLogin()->HandleReadAttribute(attributeId, buffer, maxReadLength);
+    }
+    return ret;
+}
+
+EmberAfStatus ContentApp::HandleWriteAttribute(ClusterId clusterId, chip::AttributeId attributeId, uint8_t * buffer)
+{
+    ChipLogProgress(DeviceLayer, "Read Attribute for device %s cluster %d attribute=%d)",
+                    GetApplicationBasic()->GetApplicationName(), static_cast<uint16_t>(clusterId),
+                    static_cast<uint16_t>(attributeId));
+
+    EmberAfStatus ret = EMBER_ZCL_STATUS_FAILURE;
+
+    if (clusterId == ZCL_APPLICATION_BASIC_CLUSTER_ID)
+    {
+        ret = GetApplicationBasic()->HandleWriteAttribute(attributeId, buffer);
+    }
+    if (clusterId == ZCL_ACCOUNT_LOGIN_CLUSTER_ID)
+    {
+        ret = GetAccountLogin()->HandleWriteAttribute(attributeId, buffer);
+    }
+    return ret;
+}
+
+EmberAfStatus ApplicationBasic::HandleReadAttribute(chip::AttributeId attributeId, uint8_t * buffer, uint16_t maxReadLength)
+{
+    ChipLogProgress(DeviceLayer, "ApplicationBasic::HandleReadAttribute: attrId=%d, maxReadLength=%d",
+                    static_cast<uint16_t>(attributeId), maxReadLength);
+
+    if ((attributeId == ZCL_APPLICATION_VENDOR_NAME_ATTRIBUTE_ID) && (maxReadLength == 32))
+    {
+        uint8_t bufferMemory[254];
+        MutableByteSpan zclString(bufferMemory);
+        MakeZclCharString(zclString, GetVendorName());
+        buffer = zclString.data();
+    }
+    else if ((attributeId == ZCL_APPLICATION_VENDOR_ID_ATTRIBUTE_ID) && (maxReadLength == 1))
+    {
+        *(uint16_t *) buffer = GetVendorId();
+    }
+    else if ((attributeId == ZCL_APPLICATION_NAME_ATTRIBUTE_ID) && (maxReadLength == 32))
+    {
+        uint8_t bufferMemory[254];
+        MutableByteSpan zclString(bufferMemory);
+        MakeZclCharString(zclString, GetApplicationName());
+        buffer = zclString.data();
+    }
+    else if ((attributeId == ZCL_APPLICATION_PRODUCT_ID_ATTRIBUTE_ID) && (maxReadLength == 1))
+    {
+        *(uint16_t *) buffer = GetProductId();
+    }
+    else if ((attributeId == ZCL_APPLICATION_STATUS_ATTRIBUTE_ID) && (maxReadLength == 1))
+    {
+        *buffer = (uint8_t) GetApplicationStatus();
+    }
+    else if ((attributeId == ZCL_CLUSTER_REVISION_SERVER_ATTRIBUTE_ID) && (maxReadLength == 2))
+    {
+        *buffer = (uint16_t) ZCL_APPLICATION_BASIC_CLUSTER_REVISION;
```

Comment:
This see seems very unlikely to do the right thing.....  It wouldn't even compile if it were not in a directory full of legacy code that is missing `-Wconversion`.

Reply:
this comes from bridge-app and is being replaced by AttributeAccessInterface

Reply:
Addressed in https://github.com/project-chip/connectedhomeip/pull/12969